Add Tekton PipelineRuns for SonarQube plugin and build, along with Do…

kycheng opened the pull request:
…ckerfile and plugin list

- Created two new PipelineRun configurations: `sonarqube-plugin` and `sonarqube-build` for managing SonarQube plugin builds.
- Added a Dockerfile to build the SonarQube plugin environment, which fetches plugins listed in `plugin.txt`.
- Introduced `plugin.txt` containing URLs for necessary SonarQube plugins.

These changes enhance the CI/CD process for SonarQube plugin development and deployment.

Part of 
<!-- 
  Only for standalone PRs without Jira issue in the PR title: 
    * Replace this comment with Epic ID to create a new Task in Jira
    * Replace this comment with Issue ID to create a new Sub-Task in Jira
    * Ignore or delete this note to create a new Task in Jira without a parent 
-->

Please be aware that we are not actively looking for feature contributions. We typically accept minor improvements and bug-fixes. 
Please ensure your pull request adheres to the following guidelines:
- [ ] explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] If the PR modifies or adds images, use the `run-tests.sh imagedir` command to verify the image works
- [ ] If the PR modifies or adds images, try to make sure the images run correctly on Linux

kycheng commented:
/retest

kycheng commented:
/retests

kycheng commented:
/test-all

kycheng commented:
/test-all

kycheng commented on plugin/plugins.txt:
```diff
@@ -0,0 +1,3 @@
+https://github.com/mc1arke/sonarqube-community-branch-plugin/releases/download/1.14.0/sonarqube-community-branch-plugin-1.14.0.jar
```
9.9 还需要使用14.0 没有版本更新

<img width="429" alt="image" src="https://github.com/user-attachments/assets/bfc9a7da-ec04-4acc-8905-c86e6ef0ac0b" />

kycheng commented on plugin/Dockerfile:
```diff
@@ -0,0 +1,5 @@
+
+ADD plugins.txt /tmp/plugins/
+WORKDIR /tmp/plugins
+RUN apk add --no-cache wget && wget -c -q -i ./plugins.txt
```
用于安装几个默认插件，和之前版本行为保持一致。

kycheng commented on plugin/plugins.txt:
```diff
@@ -0,0 +1,3 @@
+https://github.com/mc1arke/sonarqube-community-branch-plugin/releases/download/1.14.0/sonarqube-community-branch-plugin-1.14.0.jar
+https://github.com/vaulttec/sonar-auth-oidc/releases/download/v2.1.1/sonar-auth-oidc-plugin-2.1.1.jar
+https://github.com/spotbugs/sonar-findbugs/releases/download/v4.3.0/sonar-findbugs-plugin-4.3.0.jar
```
使用了最新版本
<img width="892" alt="image" src="https://github.com/user-attachments/assets/05953fcb-befc-4fbe-86e2-69d38311fbc4" />

l-qing commented on .tekton/plugin-pipelinerun.yaml:
```diff
@@ -0,0 +1,107 @@
+metadata:
+  name: sonarqube-plugin
+  annotations:
+    pipelinesascode.tekton.dev/on-comment: "^((/sonarqube-plugin)|(/test-all)|(/alauda-buildx-build-image)|(/test-multi.*\ alauda-buildx-build-image.*))$"
```
`alauda-buildx-build-image` 好像不太对？

l-qing commented on .tekton/plugin-pipelinerun.yaml:
```diff
@@ -0,0 +1,107 @@
+    - name: git-commit
+      value: "{{ revision }}"
+    - name: git-version
+      value: "0.1"
```
这个变量好像是多余的了

l-qing commented on .tekton/plugin-pipelinerun.yaml:
```diff
@@ -0,0 +1,107 @@
+    - name: file-list-for-commit-sha
+      value:
+        - plugin/Dockerfile
+        - plugin/plugin.txt
```
直接换成 plugin 目录是不是就行了？

l-qing commented on .tekton/plugin-pipelinerun.yaml:
```diff
@@ -0,0 +1,107 @@
+        - plugin/Dockerfile
+        - plugin/plugin.txt
+    - name: ignore-trivy-scan
+      value: true
```
这个为什么要跳过？

kycheng replied:
因为sonarqube 应用包中包含一些应用漏洞需要重新编译sonarqube 才能修复。 按之前的策略，应该通过升级来修复，暂时没有可修复版本。

在代码里加了注释。

l-qing commented on .tekton/plugin-pipelinerun.yaml:
```diff
@@ -0,0 +1,107 @@
+
+  taskRunSpecs:
+    # TODO: 待 trivy 扫描任务不依赖 root 了再移除。
+    - pipelineTaskName: image-scan
```
现在好像不需要针对该任务做特殊处理了。

l-qing commented on .tekton/sonar-pipelinerun.yaml:
```diff
@@ -0,0 +1,108 @@
+metadata:
+  name: sonarqube-build
+  annotations:
+    pipelinesascode.tekton.dev/on-comment: "^((/sonarqube-build)|(/test-all)|(/alauda-buildx-build-image)|(/test-multi.*\ alauda-buildx-build-image.*))$"
```
同上

l-qing commented on .tekton/sonar-pipelinerun.yaml:
```diff
@@ -0,0 +1,108 @@
+    - name: git-commit
+      value: "{{ revision }}"
+    - name: git-version
+      value: "0.1"
```
同上

l-qing commented on .tekton/sonar-pipelinerun.yaml:
```diff
@@ -0,0 +1,108 @@
+    - name: file-list-for-commit-sha
+      value:
+        - 9/community/Dockerfile
+        - 9/community/entrypoint.sh
```
同上

l-qing commented on .tekton/sonar-pipelinerun.yaml:
```diff
@@ -0,0 +1,108 @@
+        - 9/community/entrypoint.sh
+    
+    - name: ignore-trivy-scan
+      value: true
```
同上

l-qing commented on .tekton/sonar-pipelinerun.yaml:
```diff
@@ -0,0 +1,108 @@
+    - pipelineTaskName: image-scan
+      podTemplate:
+        securityContext:
+          runAsUser: 0
```
同上

kycheng committed:
Update Tekton PipelineRuns for SonarQube plugin and build

- Simplified comment triggers for PipelineRuns
- Removed git-version parameter
- Updated file-list-for-commit-sha to use directory paths
- Added comments explaining temporary trivy scan skip due to JAR vulnerabilities
- Removed taskRunSpecs with root user configuration for image scan task

Tongcaiyun commented:
/test-pass
